hacking/test-module.py

```diff
@@ -262,31 +259,33 @@ def rundebug(debugger, modfile, argspath, modname, module_style, interpreters):
         subprocess.call("%s %s" % (debugger, modfile), shell=True)
 
 
-def main():
-
+if __name__ == '__main__':
     options, args = parse()
-    init_plugin_loader()
     interpreters = get_interpreters(options.interpreter)
-    (modfile, modname, module_style) = boilerplate_module(options.module_path, options.module_args, interpreters, options.check, options.filename)
-
-    argspath = None
-    if module_style not in ('new', 'ansiballz'):
-        if module_style in ('non_native_want_json', 'binary'):
-            argspath = write_argsfile(options.module_args, json=True)
-        elif module_style == 'old':
-            argspath = write_argsfile(options.module_args, json=False)
-        else:
-            raise Exception("internal error, unexpected module style: %s" % module_style)
-
-    if options.execute:
-        if options.debugger:
-            rundebug(options.debugger, modfile, argspath, modname, module_style, interpreters)
-        else:
-            runtest(modfile, argspath, modname, module_style, interpreters)
-
+    argspath = write_argsfile(options.module_args)
+
+    modfile, modname, module_style = boilerplate_module(
+        options.module_path,
+        options.module_args,
+        interpreters,
+        options.check,
+        options.filename
+    )
 
-if __name__ == "__main__":
-    try:
-        main()
```

Comment:
Keeping the entry-point logic in a dedicated function is the best practice and must not be removed.